dask_expr/_expr.py

```diff
@@ -1846,10 +1848,14 @@ class Filter(Blockwise):
     operation = operator.getitem
 
     def _simplify_up(self, parent, dependents):
+        if isinstance(parent, Filter) and not isinstance(self.frame, Filter):
```

Comment:
Why is the `and not isinstance(self.frame, Filter)` necessary here?

Reply:
The explanation makes sense, a comment might be useful.

Reply:
added a comment